### PR TITLE
feat: space between CJK and English

### PR DESCRIPTION
**Pull request by iucario**
#479 

The option name also needs to be modified.

What do you think?

### CODE REVIEW

**Comment by pjkaufman** [comment]
The code change looks fine. The version change is ok, but should be reserved for the creation of a release.

The name change needs to happen, but that will require mapping from the old rule name to the new one.

**Comment by pjkaufman** [comment]
Never mind my comment about version changes. It looks like that was done automatically.

**Comment by iucario** [comment]
Currently the rule name is "Space between Chinese and English or numbers". What name should we rename it to? "Space between CJK and English or numbers"?

**Comment by pjkaufman** [comment]
I would rename it to `Space between Chinese, Japanese, or Korean and English or numbers`. This will also require renaming the class and its file and test files. We will also need to update the description of the rule to match the changes.

We will also need to map the setting data similar to https://github.com/platers/obsidian-linter/blob/master/src/main.ts#L101-L113.

**Comment by pjkaufman** [comment]
Once the file name is updated, the import for the test needs to be updated as well.

**Comment by pjkaufman** [comment]
As for mapping the data, that is to preserve the current settings to keep it from accidentally losing people's settings. It is currently pretty brittle and I am working on thinking up a better way of storing settings so that they are not brittle and will better withstand wording changes and name changes.

**Comment by pjkaufman** [comment]
As for how to map it, it depends entirely on what changes: a rule name change requires mapping the old rule name to the new rule name. Changing a description, requires mapping the old description to the new description.

Here is an example: https://github.com/platers/obsidian-linter/blob/master/src/main.ts#L101-L113

If you would like, I can add the settings mapping.

**Comment by iucario** [comment]
`npx jest --clearCache` worked! Thanks.

I changed the file name to `space-between-chinese-japanese-or-korean-and-english-or-numbers.ts`. Please help adding the settings mapping.

**Comment by pjkaufman** [comment]
I have tried to update your branch with the necessary changes for the settings mapping and one other issue I noticed, but it seems I do not have permission to do so.

You will need to add the following [here](https://github.com/iucario/obsidian-linter/blob/space-cjk/src/main.ts#L124):
``` TypeScript
const spaceBetweenChineseAndEnglishOrNumbers = this.settings.ruleConfigs['Space between Chinese and English or numbers'];
    if (spaceBetweenChineseAndEnglishOrNumbers) {
      const enabled = spaceBetweenChineseAndEnglishOrNumbers['Ensures that Chinese and English or numbers are separated by a single space. Follows these [guidelines](https://github.com/sparanoid/chinese-copywriting-guidelines)'];
      this.settings.ruleConfigs['Space between Chinese, Japanese, or Korean and English or numbers'] = {
        'Ensures that Chinese, Japanese, or Korean and English or numbers are separated by a single space. Follows these [guidelines](https://github.com/sparanoid/chinese-copywriting-guidelines)': enabled,
      };
      delete this.settings.ruleConfigs['Space between Chinese and English or numbers'];
    }
```

File: src/rules/space-between-chinese-japanese-or-korean-and-english-or-numbers.ts [hunk 1]
```diff
@@ -4,39 +4,68 @@ import dedent from 'ts-dedent';
...
-export default class SpaceBetweenChineseAndEnglishOrNumbers extends RuleBuilder<SpaceBetweenChineseAndEnglishOrNumbersOptions> {
-  get OptionsClass(): new () => SpaceBetweenChineseAndEnglishOrNumbersOptions {
-    return SpaceBetweenChineseAndEnglishOrNumbersOptions;
+export default class SpaceBetweenChineseJapaneseOrKoreanAndEnglishOrNumbers extends RuleBuilder<SpaceBetweenChineseJapaneseOrKoreanAndEnglishOrNumbersOptions> {
+  get OptionsClass(): new () => SpaceBetweenChineseJapaneseOrKoreanAndEnglishOrNumbersOptions {
+    return SpaceBetweenChineseJapaneseOrKoreanAndEnglishOrNumbersOptions;
   }
   get name(): string {
-    return 'Space between Chinese and English or numbers';
+    return 'Space between Chinese Japanese or Korean and English or numbers';
```
Comment (pjkaufman) [inline src/rules/space-between-chinese-japanese-or-korean-and-english-or-numbers.ts:16]: A grammar change:
```suggestion
    return 'Space between Chinese, Japanese, or Korean and English or numbers';
```

**Review by pjkaufman — CHANGES_REQUESTED** [review]
Other than the addition of the change to main and this suggestion here, it looks pretty good. You will likely need to do a rebase of this branch off of your master branch once you sync it with the plugin repo. That should fix the merge conflict as well.

Sorry about the delay. Thanks for putting in the work to get this added.

**Comment by pjkaufman** [comment]
Well, it looks like I was able to bypass my issue. I accidentally was trying to create a new branch instead of using the existing one. I will see about rebasing this branch and see if it fixes things.

**Review by pjkaufman — APPROVED** [review]

**Pull request merged**